### ISSUE TITLE
An error occurred while doing bitor and bitandnot

### ISSUE THREAD

**Issue by wangweiA** [ClickHouse/ClickHouse#24902]
for example :
``` sql
SELECT bitmapAndnot(bitmapOr(
    (
        SELECT groupBitmapState(toUInt32(id))
        FROM  table_a AS o
        WHERE o.mac_id IN 
        (
            SELECT o.mac_id
            FROM loc AS o
            WHERE greatCircleDistance(longitude, latitude, 120.2107, 30.20676) < 1000
        )
    ), 
    (
        SELECT groupBitmapState(toUInt32(id))
        FROM table_a AS o
        WHERE o.mac_id IN 
        (
            SELECT o.mac_id
            FROM table_l AS o
            WHERE greatCircleDistance(longitude, latitude, 120.20848, 30.206257) < 1000
        )
    )), 
    (
        SELECT groupBitmapState(toUInt32(id))
        FROM table_b AS t
        WHERE t.tagcode IN ('A0aada0200')
    ))

Query id: 706d575d-3143-44f8-8146-37408b2b7976

[ys-26] 2021.06.03 16:44:45.279151 [ 23042 ] <Fatal> BaseDaemon: ########################################
[ys-26] 2021.06.03 16:44:45.279301 [ 23042 ] <Fatal> BaseDaemon: (version 21.1.2.15 (official build), build id: E40526A12E9A8F3819A18694F6B798F10C624D5C) (from thread 23023) (query_id: 706d575d-3143-44f8-8146-37408b2b7976) Received signal Segmentation fault (11)
[ys-26] 2021.06.03 16:44:45.279365 [ 23042 ] <Fatal> BaseDaemon: Address: NULL pointer. Access: read. Address not mapped to object.
[ys-26] 2021.06.03 16:44:45.279488 [ 23042 ] <Fatal> BaseDaemon: Stack trace: 0x87b4ffd 0x7f245ac2c5f0
[ys-26] 2021.06.03 16:44:45.279577 [ 23042 ] <Fatal> BaseDaemon: 0. ? @ 0x87b4ffd in /usr/bin/clickhouse
[ys-26] 2021.06.03 16:44:45.279638 [ 23042 ] <Fatal> BaseDaemon: 1. ? @ 0xf5f0 in /usr/lib64/libpthread-2.17.so
[ys-26] 2021.06.03 16:44:45.517278 [ 23042 ] <Fatal> BaseDaemon: Checksum of the binary: F6479E166DD7E8ABFA19525726186F71, integrity check passed.

Exception on client:
Code: 32. DB::Exception: Attempt to read after eof: while receiving packet from localhost:9000
```

**Comment by qoega** [comment]
Can you reproduce it with newer version or make a reproducible example with test table that you can share with us?

**Comment by wangweiA** [comment]
OK ,  This problem has been fixed in the latest version （21.5.6.6）
